Windows TTS engine build fails on MSVC CRT mismatch from vosk-tts-rs transitives

## Symptom

Building \`kesha-engine\` with \`--features onnx,tts\` on Windows MSVC fails at link time:

\`\`\`
msvcprt.lib(MSVCP140.dll) : error LNK2005: ... already defined in libcpmt.lib(...)
LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs
fatal error LNK1169: one or more multiply defined symbols found
\`\`\`

First reproduced in PR #214 (rust-test.yml run [25059725456](https://github.com/drakulavich/kesha-voice-kit/actions/runs/25059725456/job/73410038473)) after the vosk-tts engine landed. macOS and Linux build cleanly; only Windows MSVC trips.

## Root cause

\`vosk-tts-rs 0.1.0\` pulls in a heavy transitive native-C/C++ stack via its gRPC + tokenizer features:

- \`prost-build\` → invokes \`protoc\` (added to CI in #214 commit 763cfe4)
- \`tokenizers\` → \`onig_sys\` (Oniguruma C), \`esaxx-rs\`, \`bzip2-sys\`, \`lzma-sys\`, \`zstd-sys\`

One of those native libs links against static CRT (\`/MT\`, libcmt/libcpmt) while another links against dynamic CRT (\`/MD\`, msvcrt/msvcprt). MSVC linker rejects the conflict.

Pre-existing observation: PR #214 reviewers flagged \`vosk-tts-rs\`'s heavy transitives as a binary-bloat / supply-chain concern. This is the same root: the crate ships server/gRPC modules unconditionally.

## What we did short-term

In PR #214, dropped Windows from rust-test.yml and build-engine.yml matrix temporarily so the rest of the migration can land. Released binaries continue to ship for macOS arm64 (CoreML + ONNX) and Linux x64 (ONNX); Windows x64 ONNX is not built.

## Long-term fix: vendor what we use

Spec [risk #1](../blob/main/docs/superpowers/specs/2026-04-27-vosk-ru-replacement-design.md#risks) listed vendoring as the escape hatch. Plan:

1. Copy \`model.rs\`, \`synth.rs\`, \`g2p.rs\` from \`andreytkachenko/vosk-tts-rs\` into \`rust/vendor/vosk-tts/\` (Apache 2.0 — preserve LICENSE + NOTICE).
2. Drop \`prost-build\`, \`tonic\`, \`tokenizers\`, \`reqwest\`, \`tokio\`, \`axum\`, \`hyper\` from the dep graph by removing the gRPC server / CLI / model-fetch modules entirely.
3. Replace \`tokenizers::WordPiece\` (BERT vocab loader) with a small inline reader (~50 LoC).
4. Re-enable Windows in the CI matrix.

Estimated diff: −400 transitive deps from \`Cargo.lock\`, −15-30 MB from the engine binary on platforms where DCE wasn't already eliminating them, +200 LoC vendored under \`rust/vendor/vosk-tts/\`.

Alternative: open an upstream PR feature-gating the server/CLI/tokenizers modules and wait for a release. Less code we own; weeks of latency.

## Acceptance

- \`cargo build --release --no-default-features --features onnx,tts\` succeeds on \`x86_64-pc-windows-msvc\`
- \`kesha-engine.exe\` smoke-passes \`--capabilities-json\`
- \`kesha say --voice ru-vosk-m02 "Привет мир."\` produces a valid WAV on Windows
- Re-enable Windows rows in rust-test.yml and build-engine.yml

## References

- PR #214 (vosk-ru replacement) — \`fix/replace-piper-with-vosk-ru\`
- Failing job: https://github.com/drakulavich/kesha-voice-kit/actions/runs/25059725456/job/73410038473
- Upstream: https://github.com/andreytkachenko/vosk-tts-rs (0 stars, 1 maintainer, last commit Apr 14 2026)
- Apache 2.0 license: https://github.com/andreytkachenko/vosk-tts-rs/blob/main/LICENSE

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,8 +10,19 @@ binary.
 
 ## [Unreleased]
 
+## [1.6.0] — 2026-04-30
+
+Engine release. Adds OGG/Opus voice-note output, restores Windows MSVC builds via a vendored vosk-tts crate, and tightens the Opus hot path. CLI surface is unchanged — npm consumers get the new format flag automatically once the engine binary updates.
+
 ### Added
-- **`kesha say --format ogg-opus`** — produces OGG/Opus voice notes (mono, 24 kHz @ 32 kbps by default) instead of WAV. The output file is the messenger-friendly format consumed by Telegram `sendVoice` and similar APIs. New flags `--bitrate` and `--sample-rate` tune the encoder; format is also inferred from `--out` extension (`.ogg` / `.opus` / `.oga`). All four engine paths (Kokoro plain/SSML, Vosk-TTS plain/SSML, AVSpeech) flow through the new encoder. WAV output remains the default and is byte-exact with the previous code path. (closes #223)
+- **`kesha say --format ogg-opus`** — produces OGG/Opus voice notes (mono, 24 kHz @ 32 kbps by default) instead of WAV. The output file is the messenger-friendly format consumed by Telegram `sendVoice` and similar APIs. New flags `--bitrate` and `--sample-rate` tune the encoder; format is also inferred from `--out` extension (`.ogg` / `.opus` / `.oga`). All four engine paths (Kokoro plain/SSML, Vosk-TTS plain/SSML, AVSpeech) flow through the new encoder. WAV output remains the default and is byte-exact with the previous code path. (#224, closes #223)
+
+### Changed
+- **Vendored `vosk-tts-rs`** into `rust/vendor/vosk-tts` so Windows builds compile under MSVC again — upstream's `tonic`/`prost` chain pulled in MinGW-only deps that broke the Windows engine artifact. Behaviour and the public Rust API are unchanged. (#225, closes #216)
+- npm `homepage` field now points at the project landing page (`https://drakulavich.github.io/kesha-voice-kit/`) instead of the README anchor.
+
+### Performance
+- **OGG/Opus encoder hot path:** dropped a redundant `pcm_buf.copy_from_slice` per 20 ms frame (saves N memcpys for an N-frame utterance), and right-sized the output `Vec::with_capacity` from `samples.len()` (≈6× over) to `bitrate × duration / 8 + 4 KiB`. (#226)
 
 ## [1.5.0] — 2026-04-29
 

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@drakulavich/kesha-voice-kit",
-  "version": "1.5.0",
+  "version": "1.6.0",
   "description": "Open-source voice toolkit. Speech-to-text (25 langs, ~19x faster than Whisper on Apple Silicon, ONNX fallback on CPU), text-to-speech (Kokoro + Vosk-TTS + ~180 macOS system voices, SSML), voice-activity detection, language detection (107 langs). Rust engine, OpenClaw skill.",
   "type": "module",
   "bin": {
@@ -30,7 +30,7 @@
     ]
   },
   "keshaEngine": {
-    "version": "1.5.0"
+    "version": "1.6.0"
   },
   "scripts": {
     "test": "bun test",
@@ -58,7 +58,7 @@
     "type": "git",
     "url": "git+https://github.com/drakulavich/kesha-voice-kit.git"
   },
-  "homepage": "https://github.com/drakulavich/kesha-voice-kit#readme",
+  "homepage": "https://drakulavich.github.io/kesha-voice-kit/",
   "bugs": {
     "url": "https://github.com/drakulavich/kesha-voice-kit/issues"
   },

--- a/rust/Cargo.toml
+++ b/rust/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "kesha-engine"
-version = "1.5.0"
+version = "1.6.0"
 edition = "2021"
 description = "Inference engine for Kesha Voice Kit: ASR + language detection"
 

